streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -45,15 +45,24 @@
     private long maxSegmentId = -1L;
 
     Segments(final String name, final long retentionPeriod, final int numSegments) {
+        if (numSegments < 1) {
+            throw new IllegalArgumentException("numSegments must be positive");
+        }
+        if (retentionPeriod < 1) {
+            throw new IllegalArgumentException("retentionPeriod must be positive");
+        }
         this.name = name;
         this.numSegments = numSegments;
-        this.segmentInterval = Math.max(retentionPeriod / (numSegments - 1), MIN_SEGMENT_INTERVAL);
+        this.segmentInterval = numSegments == 1 ? Long.MAX_VALUE : Math.max(retentionPeriod / (numSegments - 1), MIN_SEGMENT_INTERVAL);
```

Comment:
We actually do that for a min of 2 segments in `Windows` already: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/Windows.java#L89-L92

So how is is even possible to specify a single segment in the first place? Do you create the store manually?

Thus, I think we should keep a min segment size of two. (Or be very clear that retention time will not have any effect if only a single segment is used)